hydromt/raster.py

```diff
@@ -653,9 +653,11 @@ def origin(self) -> tuple[float, float]:
 
     def _check_dimensions(self) -> None:
         """Validate the number and order of dimensions."""
+        # get spatial dims, this also checks if these dims are present
         dims = (self.y_dim, self.x_dim)
 
-        def _check_dims(da: xr.DataArray, dims=dims) -> bool:
+        def _check_dim_order(da: xr.DataArray, dims: Tuple = dims) -> Tuple[bool, list]:
+            """Check if dimensions of a 2 or 3D array are in the right order."""
             extra_dims = [dim for dim in da.dims if dim not in dims]
             if len(extra_dims) == 1:
                 dims = tuple(extra_dims) + dims
```

Comment:
It's a really local function that is only applicable within this method. It also used the global dataset dims (this would need to move inside the function if we take it out). In what cases would this cause an issue?

Reply:
I guess thats fair. Then I think it's acceptable. I've just been burned before with the use of closures, since it's very easy for them to start returning things you don't expect. Besides they have to be parsed and allocated every time the function is called, so it's not great for performance, but this doesn't look like a critical section so , I can live with it.